src/tooling/grain_doc.md

```diff
@@ -0,0 +1,107 @@
+---
+title: grain doc
+---
+
+Grain ships with a documentation generator built directly into the compiler!
+
+You may be used to similar tools, such as `JSDoc`, that allow you to add a comment above code which describes the input and output types, along with other documentation. Generally, you'll run an external tool against your source code to generate your documentation.
+
+In Grain, we have provided this as another command via the CLI, and it inherits the full power of the compiler. For example, you won't need to specify any types in your doc comments, as the compiler will infer them. 🎉
+
+Here's a quick example of adding a doc block to your grain code:
+```gr
+/**
+ * An alias for normal syntactic array access, i.e. `array[n]`.
+ *
+ * Retrieves the element from the array at the specified index.
+ * A negative index is treated as an offset from the end of the array.
+ *
+ * @param index: The index to access
+ * @param array: The array to access
+ * @returns The element from the array
+ *
+ * @since v0.1.0
+ * @history v0.2.0: Argument order changed to data-last
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Usage
+
+Before running `grain doc`, you'll want to add doc block comments to your code. Doc block comments are started with `/**` and ended with `*/`. They must exist on the line above a type declaration or exported value. No docs will be generated for non-exported values so you can keep module internals private.
+
+You can run `grain doc <file|dir> -o <file|dir>` to generate markdown documentation for your code. As of Grain v0.5, the `grain doc` command supports directory input to recursively generate documentation for all `.gr` files.
+
```

Comment:
```suggestion

### Without doc blocks

You can even use `grain doc` without doc blocks in your code. It will still generate simple documentation for data types and exported values, such as displaying their names and type signatures.

```